modules/cluster-logging-configuration-of-json-log-data-for-default-elasticsearch.adoc

```diff
@@ -0,0 +1,111 @@
+** `openshift.labels.<key>` is the `pipeline.label.<key>` element in the `ClusterLogForwarder` whose value is used to construct the index name.
+** `kubernetes.container_name` uses the container name to construct the index name.
+* `structuredTypeName`: (string, optional) If `structuredTypeKey` is not set or its key is not present, OpenShift Logging uses the value of `structuredTypeName` as the structured type. When you use both `structuredTypeKey` and `structuredTypeName` together,  `structuredTypeName` provides a fallback index name if the key in `structuredTypeKey` is missing from the JSON log data.
+
+[NOTE]
+====
+Although you can set the value of `structuredTypeKey` to any field shown in the "Log Record Fields" topic, the most useful fields are shown in the preceding list of structure types.
+====
+
+.A structuredTypeKey: kubernetes.labels.<key> example
+
+Suppose the following:
+
+* Your cluster is running application pods that produce JSON logs in two different formats, "apache" and "google".
+* The user labels these application pods with `logFormat=apache` and `logFormat=google`.
+* You use the following snippet in your `ClusterLogForwarder` custom resource YAML file.
+
+[source,yaml]
+----
+outputDefaults:
+- elasticsearch:
+    structuredTypeKey: kubernetes.labels.logFormat <1>
+    structuredTypeName: nologformat
+pipelines:
+- inputRefs: <application>
+  outputRefs: default
+  parse: json <2>
+----
+<1> Uses the value of the key-value pair that is formed by the Kubernetes `logFormat` label.
+<2> Enables JSON log forwarding.
+
+In that case, the following structured log record goes to the `app-apache-write` index:
+
+[source,json]
+----
+{
+  "structured":{"name":"fred","home":"bedrock"},
+  "kubernetes":{"labels":{"logFormat": "apache", ...}}
+}
+----
+
+And the following structured log record goes to the `app-google-write` index:
+
+[source,json]
+----
+{
+  "structured":{"name":"wilma","home":"bedrock"},
+  "kubernetes":{"labels":{"logFormat": "google", ...}}
+}
+----
+
+.A structuredTypeKey: openshift.labels.<key> example
+
+Suppose that you use the following snippet in your `ClusterLogForwarder` custom resource YAML file.
+
+[source,yaml]
+----
+outputDefaults:
+- elasticsearch:
+    structuredTypeKey: openshift.labels.myLabel <1>
+    structuredTypeName: nologformat
+pipelines:
+ - name: application-logs
+   inputRefs:
+   - application
+   - audit
+   outputRefs:
+   - elasticsearch-secure
+   - default
+   parse: json
+   labels:
+     myLabel: myValue <2>
+----
+<1> Uses the value of the key-value pair that is formed by the OpenShift `myLabel` label.
+<2> The `myLabel` element gives its string value, `myValue`, to the structured log record.
+
+In that case, the following structured log record goes to the `app-myValue-write` index:
+
+[source,json]
+----
+{
+  "structured":{"name":"fred","home":"bedrock"},
+  "openshift":{"labels":{"myLabel": "myValue", ...}}
```

Comment:
The ellipsis is highlighted in red in the output.

Reply:
Resolved